### PR TITLE
[Feature] Implement `DualAdam` optimizer

### LINKED ISSUE #496: Support DualAdam
## Paper or Code

[Combining Adam and its Inverse Counterpart to Enhance Generalization of Deep Learning Optimizers](https://arxiv.org/abs/2603.07122
)

### PR DESCRIPTION
**Pull request by mfazrinizar**
## Problem (Why?)

Closes #496.

`pytorch-optimizer` does not yet support DualAdam, the optimizer proposed in "Combining Adam and its Inverse Counterpart to Enhance Generalization of Deep Learning Optimizers".

## Solution (What/How?)

Implemented `DualAdam` as a `BaseOptimizer` optimizer with Adam-style first and second moments, bias correction, and a linearly decaying inverse-Adam update contribution controlled by `switch_rate`.

The implementation supports the repository's standard optimizer options where they fit the algorithm: coupled weight decay by default to match the reference implementation, optional decoupled/fixed weight decay, `maximize`, sparse-gradient rejection, closure return values, complex parameters through `view_as_real`, and dynamic loading through `load_optimizer('dualadam')`.

Added registration, package export, docs API entry, README/docs table entry, changelog note, package keyword, supported-optimizer count update, a shared optimizer training recipe, and focused DualAdam tests for exact update behavior and validation paths.

## Other changes (bug fixes, small refactors)

No unrelated refactors or behavior changes.

## Notes

- Nothing so far.

## Checklist

- [x] Make sure to run `make format` before commit
- [x] My code adheres to the style guidelines of this project (`make check` shows no errors)
- [x] Both new and existing unit tests pass successfully on my local environment by running `make test`
- [x] I have made the necessary changes to the documentation

### CODE REVIEW

File: tests/test_dual_adam.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
+import pytest
```
Comment (kozistr) [inline tests/test_dual_adam.py:1]: i guess we don't need extra test script only for DualAdam optimizer cause there already has functionalities to test these! you can check out `test_optimizers.py` script!

or i can handle this after merging this PR

**Review by kozistr — APPROVED** [review]
thanks for your contribution! looks good to me overall :)

File: tests/test_dual_adam.py (hunk 1 above)
Comment (mfazrinizar) [reply tests/test_dual_adam.py:1]: I see. I'll refactor my other new optimizer implementation PRs.
